tests/pytest/test_regression.py

```diff
@@ -67,51 +68,86 @@ def get_n_processes(config):
 
 
 @pytest.mark.parametrize(
-    ("config_filename", "tag"), 
+    ("config_filename", "tag"),
     [
         ("default.yml", "{version}-debug"),
         ("baroclinic.yml", "{version}-debug"),
         ("default.yml", "{version}"),
         ("default.yml", "{version}-serialize"),
         ("restart.yml", "{version}"),
         ("model-level-coarse-graining.yml", "{version}-debug"),
-        ("pressure-level-coarse-graining.yml", "{version}-debug")
-    ]
+        ("pressure-level-coarse-graining.yml", "{version}-debug"),
+    ],
 )
 def test_regression(
-    config_filename,
-    tag,
-    image,
-    image_version,
-    reference_dir,
-    image_runner
+    config_filename, tag, image, image_version, reference_dir, image_runner
 ):
     model_image_tag = tag.format(version=image_version)
     model_image = f"{image}:{model_image_tag}"
     config = get_config(config_filename)
     run_dir = get_run_dir(model_image_tag, config)
     run_model(config, run_dir, model_image, image_runner)
-    run_name = config['experiment_name']
+    run_name = config["experiment_name"]
     run_reference_dir = os.path.join(reference_dir, run_name)
     md5sum_filename = os.path.join(run_reference_dir, MD5SUM_FILENAME)
     check_rundir_md5sum(run_dir, md5sum_filename)
-    if 'serialize' in model_image:
+    if "serialize" in model_image:
         serialize_md5sum_filename = os.path.join(
             run_reference_dir, SERIALIZE_MD5SUM_FILENAME
         )
         check_rundir_md5sum(run_dir, serialize_md5sum_filename)
     shutil.rmtree(run_dir)
 
 
-def test_callpyfort_integration(run_native, tmpdir):
-
+@pytest.fixture(scope="session")
+def emulation_run(run_native, tmpdir_factory):
     config = get_config("emulation.yml")
-    run_dir = str(tmpdir.join("rundir"))
+    rundir = tmpdir_factory.mktemp("rundir")
+    run_dir = str(rundir)
     run_native(config, run_dir)
+    return run_dir
+
+
+def test_callpyfort_integration(emulation_run):
+    run_dir = emulation_run
     assert os.path.exists(join(run_dir, "microphysics_success.txt"))
     assert os.path.exists(join(run_dir, "store_success.txt"))
 
 
+@pytest.mark.parametrize('tile', range(1, 7))
+def test_zhao_carr_diagnostics(emulation_run, regtest, tile):
+    ds = xarray.open_dataset(os.path.join(emulation_run, f"piggy.tile{tile}.nc"))
+    # print schema to regression data
+    ds.info(regtest)
```

Comment:
It might have been nice to use the existing checksum infrastructure for this test -- which also implicitly checks that the values don't change -- instead of introducing a new pattern, but perhaps the `tmpdir_factory` fixture makes that more awkward to set up?

Reply:
I didn't necessarily want to enforce bit for bit here. Also, as you mention the existing checksum infrastructure could better integrate with pytest features like regtest, `tmpdir`, etc, and avoid shelling out to system utitilies like md5sum that may or may not be present.

Reply:
That's fine -- I agree it would be nice to clean up the existing infrastructure at some point.